### ISSUE TITLE
Checking if one of jet's DB tables exists right after it's created returns false

### ISSUE THREAD

**Issue by aayla-secura** [CrocoBlock/suggestions#7797]
In `Jet_Engine_Base_DB::is_table_exists`, the result is cached, but is not updated to true when `Jet_Engine_Base_DB::create_table` is called.

So this happens:

```php
if (! $this->db->is_table_exists()) {
   $this->db->create_table();
}

assert($this->db->is_table_exists()); // fail, still returns cached false result
```

Where `$this` could be an instance of for example `Jet_Engine\Relations\Relation`.

I found this to be an issue, when I'm looking up related items for a given relation. If there aren't any related items yet, the table may not exist, and looking up `get_related_items_for_object` will log a database error. I think this is a separate issue actually: #7796 I'm getting around by checking `is_table_exists`, but this returns false if the table was just created :)

**Comment by aayla-secura** [comment]
This would also be an issue if we call `Jet_Engine\Relations\Relation::update` multiple times when the table doesn't exist yet, because internally it uses its `is_table_exists` to decide if it should create the table, and this results in the table being created multiple times and this error:

```
WordPress database error Duplicate key name 'parent_id' for query ALTER TABLE wptests_jet_rel_default ADD KEY ...
```

The second time `update` is called on the relation.

**Comment by Crocoblock** [comment]
was fixed and released in the JetEngine 3.6.0
